### DISCUSSION TITLE
Why does renderToHTML with getServerSideProps send response, return null?

### DISCUSSION THREAD

**Discussion by visum** [vercel/next.js#11525] [Help]
I would like to get the HTML result of a render of page components that use getServerSideProps, but there is a check in `renderToHTMLWithComponnets()` (packages/next/next-server/server/next-server.ts, line 996) that forces the response to be sent immediately instead of being returned.

```TypeScript
      if (html && isServerProps) {
        sendPayload(res, html, 'html', {
          private: isPreviewMode,
          stateful: true, // GSSP request
        })
        return null
      }
```

My hope was to cache the result.

Can someone help me figure out why this is desirable as designed? It seems to be on purpose, so it feels like I'm using it wrong.

**Comment by superkoh** [comment]
got the same problem, should post to issues

**Reply by visum** [reply]
Link to the issue: https://github.com/zeit/next.js/issues/11665
